lib/matplotlib/colors.py

```diff
@@ -1362,8 +1362,12 @@ def inverse(self, value):
 
     def autoscale(self, A):
         """Set *vmin*, *vmax* to min, max of *A*."""
-        self.vmin = self.vmax = None
-        self.autoscale_None(A)
+        with self.callbacks.blocked():
+            # Pause callbacks while we are updating so we only get
+            # a single update signal at the end
+            self.vmin = self.vmax = None
+            self.autoscale_None(A)
```

Comment:
Maybe just?
```suggestion
        if A.size():
            self.vmin = A.min()
            self.vmax = A.max()
```

Reply:
subclasses overwrite `autoscale_None()` to handle that themselves, so I think we need to defer to their implementations. I'd save that for another PR if someone wants to tackle it.